File: dataset_reduzido/avatar/Python/Code/codeforces_651_A.py
x , y = map ( int , input ( ).split ( ) )
print ( max ( x + y - 3 + ( ( y - x ) % 3 > 0 ) , 0 ) )
num_inp = lambda : int ( input ( ) )
arr_inp = lambda : list ( map ( int , input ( ).split ( ) ) )
sp_inp = lambda : map ( int , input ( ).split ( ) )
str_inp = lambda : input ( )

File: output/StarCoder/avatar/Python/Java/codeforces_651_A.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        String[] parts = scanner.nextLine().split(" ");
        int x = Integer.parseInt(parts[0]);
        int y = Integer.parseInt(parts[1]);
        
        System.out.println(Math.max(x + y - 3 + ((y - x) % 3 > 0), 0));
        
        int numInp() {
            return Integer.parseInt(scanner.nextLine());
        }
        
        int[] arrInp() {
            String[] elements = scanner.nextLine().split(" ");
            int[] array = new int[elements.length];
            for (int i = 0; i < elements.length; i++) {
                array[i] = Integer.parseInt(elements[i]);
            }
            return array;
        }
        
        void spInp() {
            parts = scanner.nextLine().split(" ");
        }
        
        String strInp() {
            return scanner.nextLine();
        }
        
        scanner.close();
    }
}